src/evict/evict_page.c

```diff
@@ -675,12 +675,45 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
             LF_SET(WT_REC_SCRUB);
     }
 
+    /* Acquire a snapshot if coming through eviction thread route. */
+
+    /*
+     * TODO: We were deliberately not taking a snapshot when checkpoint is active. This will ensure
+     * that point-in-time checkpoints have a consistent version of data. Remove this condition once
+     * fuzzy transaction ID based checkpoints work is merged.
+     */
+    if (FLD_ISSET(evict_flags, WT_REC_EVICTION_THREAD) && !WT_IS_HS(S2BT(session)) &&
+      !conn->txn_global.checkpoint_running) {
+        /*
+         * Eviction threads do not need to pin anything in the cache. We have a exclusive lock for
+         * the page being evicted so we are sure that the page will always be there while it is
+         * being processed. Therefore, we use snapshot API that doesn't publish shared IDs to the
+         * outside world.
+         */
+        __wt_txn_bump_snapshot(session);
+        /*
+         * If we acquired a snapshot for eviction, used forced isolation as opening and closing
+         * history store cursor for deleting history store record (as part of reconciliation) can
+         * lead to release of snapshot.
```

Comment:
The intent is good here, but I don't think it's clear. I recommend changing:

_used forced isolation as opening and closing history store cursor for deleting history store record (as part of reconciliation) can lead to release of snapshot._

To:

_force the isolation to ensure the snapshot isn't released when history store cursors are closed._

Reply:
Fixed.